test/one/profiler/test/TestProcess.java

```diff
@@ -0,0 +1,302 @@
+        cmd.add(System.getProperty("java.home") + "/bin/java");
+        cmd.add("-cp");
+        cmd.add(System.getProperty("java.class.path"));
+        if (test.debugNonSafepoints()) {
+            cmd.add("-XX:+UnlockDiagnosticVMOptions");
+            cmd.add("-XX:+DebugNonSafepoints");
+        }
+        addArgs(cmd, test.jvmArgs());
+        if (!test.agentArgs().isEmpty()) {
+            cmd.add("-agentpath:build/lib/libasyncProfiler." + libExt + "=" + substituteFiles(test.agentArgs()));
+        }
+        cmd.add(test.mainClass().getName());
+        addArgs(cmd, test.args());
+        log.log(Level.FINE, "Running " + cmd);
+        ProcessBuilder pb = new ProcessBuilder(cmd).inheritIO();
+        if (test.output()) {
+            pb.redirectOutput(createTempFile("%out"));
+        }
+        if (test.error()) {
+            pb.redirectError(createTempFile("%err"));
+        }
+        if (!test.agentArgs().isEmpty() || test.jvmArgs().contains("-Djava.library.path=build/lib")) {
+            pb.redirectOutput(createTempFile("%pout"));
+            pb.redirectError(createTempFile("%perr"));
+        }
+
+        this.p = pb.start();
+        // Give the JVM some time to initialize
+        Thread.sleep(700);
+    }
+
+    private String getExtFromFile(File file) throws IOException {
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String firstLine = reader.readLine();
+
+            if (firstLine == null) {
+                return "";
+            } else if (firstLine.contains("<!DOCTYPE html>")) {
+                return ".html";
+            } else if (firstLine.startsWith("FLR")) {
+                return ".jfr";
+            } else {
+                return "";
+            }
+        } catch (IOException e) {
+            log.log(Level.WARNING, "Error reading the file: " + e.getMessage());
+            throw e;
+        }
+    }
+
+    private void addArgs(List<String> cmd, String args) {
+        if (args != null && !args.isEmpty()) {
+            args = substituteFiles(args);
+            for (StringTokenizer st = new StringTokenizer(args, " "); st.hasMoreTokens(); ) {
+                cmd.add(st.nextToken());
+            }
+        }
+    }
+
+    private String substituteFiles(String s) {
+        Matcher m = filePattern.matcher(s);
+        if (!m.find()) {
+            return s;
+        }
+        StringBuffer sb = new StringBuffer();
+        do {
+            File f = createTempFile(m.group(1));
+            m.appendReplacement(sb, f.toString());
+        } while (m.find());
+
+        m.appendTail(sb);
+        return sb.toString();
+    }
+
+    private void addFlameGraph(File file) {
+        try {
+            List<String> cmd = new ArrayList<>();
+            cmd.add(System.getProperty("java.home") + "/bin/java");
+            cmd.add("-cp");
+            cmd.add(System.getProperty("java.class.path"));
+            cmd.add("FlameGraph");
```

Comment:
No need to call `FlameGraph` in a separate process, since it is Java code which is already on the classpath of TestRunner.

Reply:
Removed.